### PR TITLE
feat(Graph2): Tweak layouts

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage some tweaks to improve layouts (have some more in the works)...

File: packages/graph/src/graph2/graph.ts [hunk 1]
```diff
@@ -814,28 +827,16 @@ export class Graph2 extends SVGZoomWidget {
...
-                        linkDistance: this.forceDirectedLinkDistance(),
-                        linkStrength: this.forceDirectedLinkStrength()
-                    }));
+                    this.layoutAlgo(new ForceDirectedAnimated(this, this.forceDirectedOptions()));
+                    break;
+                case "ForceDirectedHybrid":
+                    const options = this.forceDirectedOptions();
+                    this.layoutAlgo(new ForceDirected(this, options)).then(() => {
+                        this.layoutAlgo(new ForceDirectedAnimated(this, options));
+                    });
```
Comment (jbrundage) [inline packages/graph/src/graph2/graph.ts:839]: 👍

File: packages/graph/src/graph2/layouts/dagre.ts [hunk 2]
```diff
@@ -59,10 +65,12 @@ export class Dagre extends Layout {
...
                 response.links.forEach(l => {
                     const e = data.edge(l.id);
+                    const sourceDist = distance(e.source.x, e.source.y, l.points[0][0] + size.width / 2, l.points[0][1] + size.height / 2);
+                    const targetDist = distance(e.target.x, e.target.y, l.points[0][0] + size.width / 2, l.points[0][1] + size.height / 2);
                     e.points = [
-                        [e.source.x, e.source.y],
+                        sourceDist < targetDist ? [e.source.x, e.source.y] : [e.target.x, e.target.y],
                         ...l.points.map(p => [p[0] + size.width / 2, p[1] + size.height / 2]),
-                        [e.target.x, e.target.y]
+                        sourceDist < targetDist ? [e.target.x, e.target.y] : [e.source.x, e.source.y]
```
Comment (jbrundage) [inline packages/graph/src/graph2/layouts/dagre.ts:73]: What's happening here? If the source position is farther from the center than the target position then swap the edge starting and ending points? Wouldn't we also need to reverse the order of the points inbetween?

File: packages/graph/src/test.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                [6, 20, "", 1],
+                [6, 21, "", 1],
+                [6, 22, "", 1],
+                [7, 22, "", 1],
+                [7, 23, "", 1],
+                [7, 24, "", 1],
+            ])
+            ;
+    }
+}
```
Comment (jbrundage) [inline packages/graph/src/test.ts:104]: Shouldn't this be a gallery sample?

**Pull request merged**